crates/aztec_backend/src/acvm_interop/proof_system.rs

```diff
@@ -17,9 +17,20 @@ impl ProofSystemCompiler for Plonk {
 
         // Add witnesses in the correct order
         // Note: The witnesses are sorted via their witness index, since we implement Ord on Witness and use a BTreeMap
+        // witness_values may not have all the witness indexes, e.g for unused witness which are not solved by the solver
+        // in that case we add a dummy value.
         let mut sorted_witness = Assignments::new();
-        for (_, value) in witness_values.iter() {
-            sorted_witness.push(*value);
+        let mut c = 0;
+        let mut i = 1;
```

Comment:
Why are we starting `i` at 1 and not 0?

Reply:
Because witness indices start with 1, not 0. The counter is first incremented and then we use it to generate the witness:
```
    // Creates a new Witness index
    fn add_witness_to_cs(&mut self) -> Witness {
        self.current_witness_index += 1;
        Witness(self.current_witness_index)
    }
```